public/app/core/components/NestedFolderPicker/NestedFolderPicker.tsx

```diff
@@ -385,3 +396,153 @@ const getStyles = (theme: GrafanaTheme2) => {
     }),
   };
 };
+
+// const listFoldersSelector = createSelector(
+//   (parentUid: string | undefined, page: number) => ({ parentUid, page }),
+//   (args) => browseDashboardsAPI.endpoints.listFolders.select(args)
+// );
+
+// export type DashboardViewItemCollection = {
+//   items: DashboardViewItem[];
+//   lastFetchedKind: 'folder' | 'dashboard';
+//   lastFetchedPage: number;
+//   lastKindHasMoreItems: boolean;
+//   isFullyLoaded: boolean;
+// };
+
+const listFolderSelector = createSelector(
+  [
+    (state: any) => state,
+    (state, parentUid: ListFolderArgs['parentUid']) => parentUid,
+    (state, parentUid: ListFolderArgs['parentUid'], page: ListFolderArgs['page']) => page,
+    (state, parentUid: ListFolderArgs['parentUid'], page: ListFolderArgs['page'], limit: ListFolderArgs['limit']) =>
+      limit,
+  ],
+  (state, parentUid, page, limit) => {
+    return browseDashboardsAPI.endpoints.listFolders.select({ parentUid, page, limit })(state);
+  }
+);
+
+function useFolderList() {
+  const requestedArgs = useRef<ListFolderArgs[] | null>(null);
+  if (requestedArgs.current === null) {
+    requestedArgs.current = [];
+  }
+
+  // TODO: propertly memoize this into a stable selector
+  const state = useSelector((rxState) => {
+    const requests = requestedArgs.current ?? [];
+
+    const pages = requests
+      .map((args) => {
+        return listFolderSelector(rxState, args.parentUid, args.page, args.limit);
+      })
+      .filter((page, index, all) => {
+        return all.findIndex((otherPage) => otherPage.requestId === page.requestId) === index;
+      });
```

Comment:
Step 2: Then, when the store updates, loop over the requests and select the data from RTKQ to use